scripts/2_create.sql

```diff
@@ -25,37 +25,36 @@ CREATE TABLE aws_hostnames
 
 -- Notice "PARTITION BY" below.  We're using built-in partitioning from Postgres 10+
 -- See https://blog.timescale.com/scaling-partitioning-data-postgresql-10-explained-cd48a712a9a1/
--- Postgres 11 has many updates for partitioning, and when we can use Postgres 11, we can uncomment
--- the FOREIGN KEY bits.  See:  https://pgdash.io/blog/partition-postgres-11.html
+-- Postgres 11 has many updates for partitioning (see https://pgdash.io/blog/partition-postgres-11.html)
+-- that we are taking advantage of, such as automatic index propagation to child tables.
 
 CREATE TABLE aws_events_ips_hostnames
 (
     ts TIMESTAMP NOT NULL,
     is_public BOOLEAN NOT NULL,
     is_join BOOLEAN NOT NULL,
     aws_resources_id VARCHAR NOT NULL,
-    -- FOREIGN KEY (aws_resources_id) REFERENCES aws_resources (id),
+    FOREIGN KEY (aws_resources_id) REFERENCES aws_resources (id),
     aws_ips_ip INET NOT NULL,
-    -- FOREIGN KEY (aws_ips_ip) REFERENCES aws_ips (ip),
-    aws_hostnames_hostname VARCHAR
-    -- ,
-    -- FOREIGN KEY (aws_hostnames_hostname) REFERENCES aws_hostnames (hostname)
+    FOREIGN KEY (aws_ips_ip) REFERENCES aws_ips (ip),
+    aws_hostnames_hostname VARCHAR,
+    FOREIGN KEY (aws_hostnames_hostname) REFERENCES aws_hostnames (hostname)
 )
 PARTITION BY
     RANGE
 (
         ts
 );
 
--- And we'll make sure there's an index right away:
--- Postgres 11 has many updates for partitioning, and when we can use Postgres 11, we can uncomment
--- the FOREIGN KEY bits.  See:  https://pgdash.io/blog/partition-postgres-11.html
+-- And we'll make sure there's an index right away that Postgres 11 will automatically propagate to child tables:
+-- (We use aws_ips_ip and timestamp as it is expected these will be the most frequently used in queries)
 
--- CREATE INDEX
--- IF NOT EXISTS aws_events_ips_hostnames_aws_ips_ip_ts_idx ON aws_events_ips_hostnames USING BTREE
--- (aws_ips_ip, ts);
+CREATE INDEX
+IF NOT EXISTS aws_events_ips_hostnames_aws_ips_ip_ts_idx ON aws_events_ips_hostnames USING BTREE
```

Comment:
Request: Can we create an index on hostname/ts? I think vuln scanning team will be searching by this dimension as well.

Reply:
We shouldn't need all of those indexes on the join table. If we're querying by a specific dimension then we should query the table that records that dimension and then join it back to the details.

We really shouldn't need additional indexes on the join table (which is effectively a view at this point) because the primary keys and foreign keys are already indexed.  Additionally, the primary keys of the other fact and dimension tables are indexed. Adding additional indexes somewhere should be treated like a code smell in 4NF or 5NF schema design.

Reply:
Right, agree on the unnecessary indices.  I'll remove the `CREATE INDEX` and sub in a code comment.